File: app/src/main/java/com/example/mnowak/cirriculumvitae/feature/moreInfo/fragment/skills/SkillsFragment.java
package com.example.mnowak.cirriculumvitae.feature.moreInfo.fragment.skills;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.mnowak.cirriculumvitae.R;
import com.example.mnowak.cirriculumvitae.model.SkillsViewModel;

import java.io.Serializable;
import java.util.Arrays;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class SkillsFragment extends Fragment {
    private static final String ARG_SKILLS_MODEL = "skillsModel";

    @BindView(R.id.recyclerView)
    RecyclerView recyclerView;

    private List<SkillsViewModel> skills;

    public static SkillsFragment newInstance(List<SkillsViewModel> skills) {
        Bundle args = new Bundle();
        args.putSerializable(ARG_SKILLS_MODEL, skills.toArray(new SkillsViewModel[]{}));
        SkillsFragment fragment = new SkillsFragment();
        fragment.setArguments(args);
        return fragment;
    }


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        readArguments();
    }

    private void readArguments() {
        if (getArguments() != null) {
            Serializable serializableActivities = getArguments().getSerializable(ARG_SKILLS_MODEL);
            if (serializableActivities instanceof SkillsViewModel[]) {
                skills = Arrays.asList((SkillsViewModel[]) serializableActivities);
            }
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_recycler_view, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        ButterKnife.bind(this, view);
        recyclerView.setAdapter(new SkillsRecyclerViewAdapter(skills));
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
    }
}

File: app/src/main/java/com/mnowak/curriculumvitae/feature/moreInfo/fragment/skills/SkillsFragment.kt
package com.mnowak.curriculumvitae.feature.moreInfo.fragment.skills

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.mnowak.curriculumvitae.R
import com.mnowak.curriculumvitae.di.viewModel.ViewModelFactory
import com.mnowak.curriculumvitae.feature.moreInfo.fragment.skills.list.SkillsAdapter
import dagger.android.support.DaggerFragment
import javax.inject.Inject

class SkillsFragment : DaggerFragment() {

    @Inject
    lateinit var viewModelFactory: ViewModelFactory
    lateinit var recyclerView: RecyclerView

    private val adapter = SkillsAdapter(this)

    private val viewModel: SkillsViewModel by lazy {
        ViewModelProviders.of(this, viewModelFactory).get(SkillsViewModel::class.java)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        recyclerView = inflater.inflate(R.layout.fragment_recycler_view, container, false) as RecyclerView
        return recyclerView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupRecyclerView()
        observe()
    }

    private fun setupRecyclerView() {
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
    }

    private fun observe() {
        viewModel.skills.observe(viewLifecycleOwner, Observer {
            adapter.submitList(it)
        })
    }

    companion object {

        fun newInstance(): SkillsFragment {
            return SkillsFragment()
        }
    }
}
